pkg/ottl/compare.go

```diff
@@ -12,13 +12,42 @@ import (
 	"golang.org/x/exp/constraints"
 )
 
+// ValueComparator defines methods for comparing values using the OTTL comparison rules
+// (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/LANGUAGE.md#comparison-rules)
+type ValueComparator interface {
+	// Equal compares two values for equality, returning true if they are equals
+	// according to the OTTL comparison rules.
+	Equal(a any, b any) bool
+	// NotEqual compares two values for equality, returning true if they are different
+	// according to the OTTL comparison rules.
+	NotEqual(a any, b any) bool
+	// Less compares two values, returning true if the first value is less than the second
+	// value, using the OTTL comparison rules.
+	Less(a any, b any) bool
+	// LessEqual compares two values, returning true if the first value is less or equal
+	// to the second value, using the OTTL comparison rules.
+	LessEqual(a any, b any) bool
+	// Greater compares two values, returning true if the first value is greater than the
+	// second value, using the OTTL comparison rules.
+	Greater(a any, b any) bool
+	// GreaterEqual compares two values, returning true if the first value is greater or
+	// equal to the second value, using the OTTL comparison rules.
+	GreaterEqual(a any, b any) bool
+	// compare is a private method that compares two values using the grammar compareOp,
+	// it also restricts custom implementations outside of this package.
+	compare(a any, b any, op compareOp) bool
+}
+
+// ottlValueComparator is the default implementation of the ValueComparator
+type ottlValueComparator struct{}
```

Comment:
Can you add the `var _ = (ValueComparator)(nil)` thing that enforces this type is a `ValueComparator`?

Reply:
I think in this case we probably don't need that, the `NewValueComparator() ValueComparator { return &ottlValueComparator{} }` is already ensuring it, if `ottlValueComparator` does not satisfy the interface, compilation would break.